File: Dataset/Leetcode/train/38/536.cpp
 class Solution {
public:
    string XXX(int n) {
        if(n == 1) return "1";
        string ans = "1";
        for(int i = 1; i < n; i++)
            ans = getTarget(ans);
        return ans;
    }

    string getTarget(string s){
        string ans;
        char tmp = s[0];
        int cnt = 0;
        for(int i = 0; i < s.size(); i++){
            if(s[i] != tmp){
                ans += to_string(cnt);
                ans += tmp;
                cnt = 1;
                tmp = s[i];
            }
            else ++cnt;
        }
        if(cnt){
            ans += to_string(cnt);
            ans += tmp;
        }
        return ans;
    }
};

File: Dataset/Leetcode/train/6/536.java
 class Solution {
    public String XXX(String s, int numRows) {
        if(numRows == 1){
            return s;
        }
        StringBuilder sb = new StringBuilder();
        int bi = (2 * numRows - 2);
        for(int i = 1; i <= Math.min(numRows, s.length()); i++){
            int k = (s.length() - i) / bi;
            for(int j = 0; j <= k; j++){
                if(j > 0 && i != 1 && i != numRows){
                    sb.append(s.charAt(i + j * bi - 2 * i + 1));
                }
                sb.append(s.charAt(i + j * bi - 1));
            }
            if(i != 1 && i != numRows && i + (k + 1) * bi - 2 * i + 1 < s.length()){
                sb.append(s.charAt(i + (k + 1) * bi - 2 * i + 1));
            }
        }
        return sb.toString();
    }
}

